packages/mui-joy/src/Box/Box.test.js

```diff
@@ -223,31 +217,44 @@ describe('Joy <Box />', () => {
 
       const { container } = render(
         <CssVarsProvider theme={theme}>
-          <Box sx={{ letterSpacing: 'md' }} />
+          <Box sx={{ lineHeight: 'md' }} />
         </CssVarsProvider>,
       );
 
       expect(container.firstChild).toHaveComputedStyle({
-        letterSpacing: '1.328px',
+        lineHeight: '24px',
       });
     });
+  });
+});
 
-    it('lineHeight', function test() {
-      const isJSDOM = /jsdom/.test(window.navigator.userAgent);
-
-      if (isJSDOM) {
-        this.skip();
-      }
+// eslint-disable-next-line mocha/max-top-level-suites
+describe('theme.unstable_sx', () => {
+  const theme = extendTheme({
+    colorSchemes: {
+      light: {
+        palette: {
+          primary: {
+            500: 'rgb(0, 0, 255)',
+          },
+        },
+      },
+    },
+  });
 
-      const { container } = render(
-        <CssVarsProvider theme={theme}>
-          <Box sx={{ lineHeight: 'md' }} />
-        </CssVarsProvider>,
-      );
+  it('bgcolor', () => {
+    expect(theme.unstable_sx({ bgcolor: 'primary.500' })).to.deep.equal({
+      // TODO fixme
+      // backgroundColor: 'var(--joy-palette-primary-500)',
+      backgroundColor: 'primary.500',
```

Comment:
This probably fails because it is not used inside `CssVarsProvider`. The `CssVarsProvider` is required when using @mui/joy: https://mui.com/joy-ui/getting-started/usage/#quickstart. cc @siriwatknp is this correct?

Reply:
yes, you are correct.

The `theme` output from `extendTheme()` does not have `theme.vars` and `theme.palette.*` yet. That's why `theme.unstable_sx` returns `primary.500` because it could not find the palette.

Using ThemeProvider should be enough because it attaches `vars` and use `light` color scheme as a palette.

Reply:
All other tests already use the `CssVarsProvider`, I'd suggest adding it to this test too.

Reply:
I push a change, I moved these tests to `extendTheme`, as it looked more related to it. I also wrap the tests with `CssVarsProvider` and the results look as expected now.

Reply:
Why is `CssVarsProvider` mutating the theme?

Reply:
I would expect that the theme would add `vars` and maybe point them to the plain values before the CssVarsProvider is used. This would be safest. The palette could be generated the same way too. I know we have it for `defaultTheme`, maybe we can just add this logic to `extendTheme` too. I was thinking about this when creating the theme for Material You, but for now used the same approach in Joy UI.

Reply:
In the code: https://github.com/mui/material-ui/blob/33a0e1c2ebd7575f694638559f42eb9a9d545e60/packages/mui-system/src/cssVars/createCssVarsProvider.js#L154-L158

It seems that `theme.vars` is defined with params that aren't specific to the theme provider. This what put me on the track of, we should be able to do:

```jsx
export const theme = extendTheme({ /* … */ });
```

```jsx
import theme from './myTheme';

{
  color: theme.vars.palette.primary.main,
},
```

Pretty much like https://vanilla-extract.style/documentation/api/create-theme/.

> for now used the same approach in Joy UI

👍 to keep these in sync

Reply:
I will draft a PR for this to see how the changes would look like.